### PR TITLE
Add debugging instructions to backend README

### PR DESCRIPTION
**Pull request by timbonicus**
@mtlewis and I were struggling with loggers, found that you can attach to the node process. I didn't see this documented anywhere, thought it might be generally useful.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by OrkoHunter — APPROVED** [review]
Noice

File: packages/backend/README.md [hunk 1]
```diff
@@ -39,6 +39,21 @@ You can also, instead of using dummy values for a huge number of environment var
...
+
+To debug the backend in [Visual Studio Code](https://code.visualstudio.com/):
+
+- Enable Auto Attach (⌘ + Shift + P > Toggle Auto Attach > Only With Flag)
+- Open a VSCode terminal (Control + `)
+- Run the backend from the VSCode terminal:
+
+```
+$ cd packages/backend
+$ yarn backstage-cli backend:dev --inspect
```
Comment (freben) [inline packages/backend/README.md:54]: would probably be more familiar, I think, to just make this `yarn start-backend --inspect` from the root - or does not the flag get passed along then?

**Review by freben — APPROVED** [review]

File: packages/backend/README.md (hunk 1 above)
Comment (timbonicus) [reply packages/backend/README.md]: I tried this initially and had the debugger trying to latch on to intermediate steps (e.g. `yarn start-backend` gets a debugger, `backstage-cli backend:dev` gets another). That must have been with a different auto attach mode though, seems to work as expected when I tried it now. Updated!

Comment (freben) [reply packages/backend/README.md]: Yeah that mode was smart - i didn't know if it, neat

**Pull request merged**